### PR TITLE
Generate User API

### PR DESCRIPTION
**Pull request by trmi1012**
Added template.yml for UserAPI

Adjustments in ModelAPI.stencil to correctly generate UserApi requests

### CODE REVIEW

File: Templates/Kotlin/Common/Includes/ModelAPI.stencil [hunk 1]
```diff
@@ -34,7 +31,7 @@ object {{ type }}API {
 
     private val service: {{ type|upperCamelCase }}Service by lazy {
         Retrofit.Builder()
-            .client(OkHttpClient.Builder().addInterceptor(InterceptorUtils.getInterceptor("application/json", "application/json")).build())
+            .client(OkHttpClient.Builder().addInterceptor(InterceptorUtils.getInterceptor({% if type == "GetTerms" %}"application/vnd.api+json", "application/vnd.api+json"{% else %}"application/json", "application/json"{% endif%})).build())
```
Comment (EmDee) [inline Templates/Kotlin/Common/Includes/ModelAPI.stencil:34]: @trmi1012 This should be handled more generically. However, I don't see that the request of `GetTerms` should set the `content-type` to `application/vnd.api+json`. Why is this needed? 

With the changes from https://github.com/pace/SwagGen/pull/22 we should be able to access the content type of requests more generically.

Comment (trmi1012) [reply Templates/Kotlin/Common/Includes/ModelAPI.stencil:34]: I've seen the changes in #22 but could not access content type yet. Since we'd have to update all generated APIs because it is currently set statically for every request I suggest doing this in another issue. User Api stuff and App MR's already merged and working of course.

Comment (EmDee) [reply Templates/Kotlin/Common/Includes/ModelAPI.stencil:34]: Does the `GetTerms` explicitly need the `application/vnd.api+json` content type though? It didn't work with just `application/json`?

Comment (trmi1012) [reply Templates/Kotlin/Common/Includes/ModelAPI.stencil:34]: As content type application/json is fine. But for the access header it is required otherwise getting 415

Comment (EmDee) [reply Templates/Kotlin/Common/Includes/ModelAPI.stencil:34]: Ok, we will do this in a follow-up then.
